scripts/quests/otherAreas/X_Marks_The_Spot.lua

```diff
@@ -0,0 +1,128 @@
+-----------------------------------
+-- X Marks the Spot
+-----------------------------------
+-- Log ID: 4, Quest ID: 65
+-----------------------------------
+-- CoP Mission 2-5 Ancient Vows Completed
+-- !addmission 6 258
+-----------------------------------
+-- Tavnazian Safehold       !zone 26
+-- Despachiaire             !pos 111.2090 -40.0148 -85.4810
+-- Parelbriaux              !pos 113.701 -41 42.3653
+-- Odeya                    !pos 83.3227 -34.25 -70.0384
+-- Phomiuna Aqueducts       !zone 27
+-- Hidden Hallway 2nd Gate  !pos 138.1027 -24 60.3209
+-----------------------------------
+
+local quest = Quest:new(xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.X_MARKS_THE_SPOT)
+
+quest.reward =
+{
+    gil = 4000,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+            player:getCurrentMission(xi.mission.log_id.COP) > xi.mission.id.cop.ANCIENT_VOWS
+        end,
+
+        [xi.zone.TAVNAZIAN_SAFEHOLD] =
+        {
+            ['Despachiaire'] = quest:progressEvent(144),
+
+            onEventFinish =
+            {
+                [144] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.TAVNAZIAN_SAFEHOLD] =
+        {
+            -- Will repeat until the quest is completed.
+            ['Despachiaire'] = quest:progressEvent(144),
```

Comment:
Is this confirmed to be blocking? Not sure what else he could be handling, and would be fine assuming all of his other interactions are progress

Reply:
Rewatching the capture it does not block. However, it is one of the scenes he cycles through depending on what you have active, and this cs will get cycled in until you complete the quest. 

I'll need to recode this. However, I'm unsure how to go about making sure it still cycles in other scenes available.

Reply:
Use `quest:event(x)`

If the other quests he is involved in are properly coded using the priority method, the cycle will happen automatically.

:progressEvent() means the event takes priority over other events of lower priority. :event() has lower priority, for instance.

If 2 events have the same priority, the cycling happens automatically.